fastcrawler/test_utils.py

```diff
@@ -0,0 +1,62 @@
+from fastcrawler.engine import AioHttpEngine
+
+
+class TestServer:
+    def __init__(self):
+        self.responses = {
+            "GET": {"message": "GET request received"},
+            "POST": {"message": "POST request received"},
+            "PUT": {"message": "PUT request received"},
+            "DELETE": {"message": "DELETE request received"},
+        }
+
+    def get_response(self, method, data=None):
+        response = self.responses.get(method)
+        if data:
+            response["data"] = data
+        return response
```

Comment:
When you're mocking something, there's 3 things you should really respect
1. The mocked version and real version MUST have same input/output. this engine is so different.
2. All method should be mocked, on the interface
3. You should "understand" why you're mocking something. The reason is to actually not "SEND REQUEST" to any server right? that means we should be also mocking responses! what does this mean? it means user should be able to redirect responses to url, say hey I want "/user/{user_id}" to return this HTML. your mock is not supporting that, which make it useless.